request_handler.go

```
@@ -24,100 +25,66 @@ type Profile struct {
-	updateInfo := UpdateInfo{}
-	updateInfo.Username = p.Username
-	updateInfo.Email = p.Email
-	updateInfo.Description = p.Description
-	jsonData, err := json.Marshal(updateInfo)
+	request, err := http.NewRequest("GET", apiUrl,nil)
+	Cookie, err := r.Cookie("cookie")
 	if err != nil {
 		log.Println(err)
 	}
-	Data := string(jsonData)
-	// Encode the information to byte payload
-	payload := strings.NewReader(Data)
-	request, err := http.NewRequest("PUT", link, payload)
-	request.Header.Add("Content-Type", "application/json")
-	_, err = client.Do(request)
-	client.CloseIdleConnections()
+	request.Header.Add("Cookie", strings.TrimPrefix(Cookie.Value, "cookie="))
+	resp, err := client.Do(request)
 	if err != nil {
 		log.Println(err)
 	}
-
-	return err
-}
-
-// Preload the information by fetching data from backend
-func readFromBackend(userid string) (*Profile, error) {
-	apiUrl := "http://localhost:8080/v1/accounts/@me?token=" + token
-	res, err := http.Get(apiUrl)
-	if err != nil {
-		log.Println(err)
-	}
-	defer res.Body.Close()
-	bodyBytes, err := ioutil.ReadAll(res.Body)
-	//var responseInfo Response
+	defer resp.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	// Extract data from response and turn to pageInfo type
 	var pageInfo Profile
 	err = json.Unmarshal(bodyBytes, &pageInfo)
 	if err != nil {
 		log.Println(err)
 	}
-	// Extract the data from http response and generate data to render backend pages
-	//data := responseInfo.Message
-	//
-	//err = json.Unmarshal([]byte(data), &pageInfo)
-	//if err != nil {
-	//	log.Fatal(err)
-	//}
+	fmt.Println(pageInfo)
 	return &pageInfo, err
 }
 
-//Function to read public information without login and render the web pages
+// Function to read public information without login and render the web pages
 func readFromPublic(username string) (*PublicInfo, error) {
 	link := "http://localhost:8080/v1/accounts/" + username
 	resp, err := http.Get(link)
 	if err != nil {
-		log.Fatal(err)
+		log.Println(err)
```

It shouldn't just swallow the error